script/learncpan.pl

```diff
@@ -0,0 +1,162 @@
+use strict;
+use warnings;
+use open qw/ :std :utf8 /;
+use Mojo::UserAgent;
+use Text::CSV qw( csv );
+
+my %suffixes = (
+    1  => 'st',
+    2  => 'nd',
+    3  => 'rd',
+    21 => 'st',
+    22 => 'nd',
+    23 => 'rd'
+);
+
+# Get all tutorials that are already listed
+my $tutorials = csv(
+    in      => "../learncpan/tutorials.csv",
+    headers => "auto"
+);
+
+my %dists;
+foreach my $tutorial (@$tutorials) {
+    $dists{ lc $tutorial->{distribution} } = 1;
+}
+
+my $ua = Mojo::UserAgent->new;
+
+my @rows;
+my @days = ( 1 .. 25 );
+
+# 2000
+foreach my $day (@days) {
+
+    my $url = get_url( 2000, $day );
+    my $dom = get_dom($url);
+    next unless $dom;
+
+    my $h1     = $dom->at('h1');
+    my $p      = $h1->next;
+    my $title  = $p->text;
+    my $module = $title;
+
+    $title  =~ s/,//g;
+    $module =~ s/\s-\s.+//;
+    $module =~ s/::/-/g;
+
+    if ( $module =~ m/\s/ ) {
+        print "Not a module: $module at $url \n";
+    }
+    elsif ( !$dists{ lc $module } ) {
+        push @rows, [ $module, $url, $title ];
+
+    }
+}
+
+foreach my $year ( 2001 .. 2004 ) {
+    foreach my $day (@days) {
+
+        my $url = get_url( $year, $day );
+        get_articles( $url,
+            { title_selector => 'title', module_selector => 'div.modtitle' } );
+    }
+}
+
+
+# There is no sure way of identyfing the main module in the article for these years from the HTML
+foreach my $year ( 2005 .. 2010 ) {
+    foreach my $day (@days) {
+
+        my $url = get_url( $year, $day );
+        my $dom = get_dom($url);
+        next unless $dom;
+        my $title = $dom->at('title');
+
+        push @rows, [ 'TODO', $url, $title ];
+    }
+}
+
+foreach my $year ( 2011 .. 2020 ) {
```

Comment:
I think we can skip 2020 as it was (if I'm correct) a collection of highlights from previous years.

Reply:
+1